### PR TITLE
Allow the user to specify a YaccKind on the command line.

### PR DESCRIPTION
**Pull request by ltratt**
This allows one to do e.g.:

  lrpar -y eco lang.l lang.y input.lang

While here, slightly tidy up the usage function and (hoho) its usage.

### CODE REVIEW

**Comment by snim2** [comment]
Do you want to kick that Travis build?

File: src/main.rs [hunk 1]
```diff
@@ -85,14 +85,31 @@ fn main() {
...
-        return;
+    }
+
+    let yacckind = match matches.opt_str("y") {
+        None => YaccKind::Original,
+        Some(s) => {
+            match &*s.to_lowercase() {
+                "original" => YaccKind::Original,
+                "eco" => YaccKind::Eco,
+                _ => usage(prog, &format!("Unknown yacckind '{}'.", s))
```
Comment (snim2) [inline src/main.rs]: If you call this a "Yacc variant" in the help text then it would make sense to use "variant" rather than "kind" in the error message.

**Comment by snim2** [comment]
This looks sensible to me. I tend to prefer docopt for argument parsing, but if you are only expecting to have a small number of CLI arguments it may not be worth the hassle.

**Comment by ltratt** [comment]
Personally I think these parsing libraries are all too complex, so I'm sticking with the simplest one my little brain can cope with!

File: src/main.rs (hunk 1 above)
Comment (ltratt) [reply src/main.rs]: Good point. Fixed in latest commit (hopefully).

**Comment by ltratt** [comment]
If that latest commit works for you, it's probably squashing time.

**Comment by snim2** [comment]
Yes, please squash!

**Comment by ltratt** [comment]
OK, squashed.

**Pull request merged**